certstore_windows.go

```diff
@@ -358,62 +383,112 @@ func (wpk *winPrivateKey) Public() crypto.PublicKey {
 
 // Sign implements the crypto.Signer interface.
 func (wpk *winPrivateKey) Sign(rand io.Reader, digest []byte, opts crypto.SignerOpts) ([]byte, error) {
-	if wpk.capiProv != 0 {
-		return wpk.capiSignHash(opts.HashFunc(), digest)
-	} else if wpk.cngHandle != 0 {
-		return wpk.cngSignHash(opts.HashFunc(), digest)
+	if wpk.cngHandle != 0 {
+		return wpk.cngSignHash(opts, digest)
 	} else {
 		return nil, errors.New("bad private key")
 	}
 }
 
+type bcryptPkcs1PaddingInfo struct {
+	pszAlgID *uint16
+}
+
+type bcryptPssPaddingInfo struct {
+	pszAlgID *uint16
+	cbSalt   uint64
```

Comment:
[`BCRYPT_PSS_PADDING_INFO`][1]'s `cbSalt` field is a `ULONG`. In Win32 a ULONG is an "unsigned LONG", but a `LONG` is [always 32-bits in Windows][2]. Then I think this would be correct to declare this as a `uint32`.

[1]: https://docs.microsoft.com/en-us/windows/win32/api/bcrypt/ns-bcrypt-bcrypt_pss_padding_info
[2]: https://docs.microsoft.com/en-us/windows/win32/winprog/windows-data-types#long